src/app/modules/convai-checker/perspectiveapi.service.ts

```diff
@@ -93,11 +96,7 @@ export class PerspectiveApiService {
                       + ' Defaulting to current hosted address');
       }
 
-      const headers = new HttpHeaders();
-      headers.append('Content-Type', 'application/json');
-
-      return this.httpClient.post(
-        serverUrl + '/check', data, {headers});
+      return this.sendRequest(data, `${serverUrl}/check`, 'checkText');
```

Comment:
sendRequest has generic parameters T and R, you should specify them here like this:

this.sendRequest<AnalyzeCommentData, AnalyzeCommentResponse>(data, `${serverUrl}/check`, 'checkText');

(Also do this in the other places sendRequest() is called).

Reply:
Done.